agent/src/agent_win.h

```diff
@@ -17,30 +19,121 @@ namespace sdk {
+    DWORD ConnectPipe();
+
+    // Resets this connection object to listen for a new Google Chrome browser.
+    DWORD ResetInternal(std::string& pipename, bool is_first_pipe);
+
+    // Cleans up this connection object so that it can be reused with a new
+    // Google Chroem browser instance.  The handles assocated with this object
+    // are not closed.  On return, this object is neither connected nor
+    // listening and any buffer used to hold browser messages are cleared.
+    void Cleanup();
+
+    // Queues a read on the pipe to receive a message from Google Chrome.
+    // ERROR_SUCCESS, ERROR_IO_PENDING, and ERROR_MORE_DATA are successful
+    // return values. Other values represent an error with the connection.
+    // If `reset_cursor` is true the internal read buffer cursor is reset to
+    // the start of the buffer, otherwise it is unchanged.
+    DWORD QueueReadFile(bool reset_cursor);
+
+    // Called when data from Google Chrome is available for reading from the
+    // pipe. ERROR_SUCCESS and ERROR_MORE_DATA are both successful return
+    // values.  Other values represent an error with the connection.
+    DWORD OnReadFile(BOOL success, DWORD count);
+
+    // Calls the appropriate method the handler depending on the message
+    // received from Google Chrome.
+    DWORD CallHandler();
+
+    // Fills in the browser_info_ member of this Connection.  Assumes
+    // IsConnected() is true.
+    DWORD BuildBrowserInfo();
+
+    // The handler to call for various agent events.
+    AgentEventHandler* handler_ = nullptr;
+
+    // Members used to communicate with Google Chrome.
+    HANDLE handle_ = INVALID_HANDLE_VALUE;
+    OVERLAPPED overlapped_;
+
+    // True if this connection is assigned to a specific Google Chrome browser,
+    // otherwise this connection is listening for a new browser.
+    bool is_connected_ = false;
+
+    // Information about the Google Chrome browser process.
+    BrowserInfo browser_info_;
+
+    // Members used to read messages from Google Chrome.
+    std::vector<char> buffer_;
+    char* cursor_ = nullptr;
+    DWORD read_size_ = 0;
+    DWORD final_size_ = 0;
+  };
+
+  // Returns handles that can be used to wait for events from all Connection
+  // objects managed by this agent.
+  void GetHandles(std::vector<HANDLE>& wait_handles) const;
+
+  // Performs a clean shutdown of the agent.
   void Shutdown();
 
+  // Name used to create the pipes between the agent and Google Chrome browsers.
   std::string pipename_;
-  HANDLE hPipe_ = INVALID_HANDLE_VALUE;
+
+  // A list of pipes to already connected Google Chrome browsers.
+  // The first kNumPipeInstances pipes in the list correspond to listening
+  // pipes.
+  std::vector<std::unique_ptr<Connection>> connections_;
 };
 
 }  // namespace sdk
 }  // namespace content_analysis
 
-#endif  //q CONTENT_ANALYSIS_AGENT_SRC_AGENT_WIN_H_
\ No newline at end of file
+#endif  // CONTENT_ANALYSIS_AGENT_SRC_AGENT_WIN_H_
```

Comment:
There is a `no newline at the end of the file` from GitHub here. Is there some setting we can add so that newlines are consistent across the SDK without needing individual contributors to tweak their editor settings?